cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -169,32 +183,6 @@ where
 		}
 	}
 
-	/// Checks the execution receipt from the executor peers.
-	///
-	/// TODO: invoke this once the external ER is received.
-	#[allow(unused)]
-	async fn on_execution_receipt_received(
-		&mut self,
-		_execution_receipt: ExecutionReceipt<<Block as BlockT>::Hash>,
-	) {
-		// TODO: validate the Proof-of-Election
-
-		// TODO: check if the received ER is same with the one produced locally.
-		let same_with_produced_locally = true;
-
-		if same_with_produced_locally {
-			// TODO: rebroadcast ER
-		} else {
-			// TODO: generate a fraud proof
-			let fraud_proof = FraudProof { proof: StorageProof::empty() };
-
-			// TODO: gossip the fraud proof to farmers
-			self.overseer_handle
-				.send_msg(CollationGenerationMessage::FraudProof(fraud_proof), "SubmitFraudProof")
-				.await;
-		}
-	}
-
 	async fn produce_candidate(
 		mut self,
```

Comment:
Probably, I simply followed the original pattern of cumulus. Since we always use an executor clone in the closure following, so doesn't make a difference whether the instance is consumed as it's a clone anyway.

Reply:
The only reason I asked this is because it is cloned twice just because of this signature. And generally `mut self` is an indicator of a very specific behavior, which is not used or relied on here.

Reply:
`mut self` is required here due to `&mut self` being used in `produce_candidate` of `ParachainConsensus` trait. Since this piece of code will be removed eventually, we could ignore it for now.